ArduPlane/RC_Channel.cpp

```diff
@@ -123,6 +123,20 @@ void RC_Channel_Plane::do_aux_function(const aux_func_t ch_option, const aux_swi
         do_aux_function_change_mode(Mode::Number::TAKEOFF, ch_flag);
         break;
 
+    case AUX_FUNC::AVOID_ADSB:
+
+        // enable or disable AP_Avoidance
+        if (ch_flag == HIGH && plane.adsb.enabled() ) {
```

Comment:
When this is moved into the library, create a method to handle the switch being enabled, rather than putting all of this in-line in the switch statement.  This should allow thecode to be cleaner as you can early-exit from the routine.

This may stop things like the redundant check for adsb-not-enabled-while-channel-flag-high (on line 132) from creeping in.